hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/StreamSourceStage.java

```diff
@@ -49,6 +49,11 @@
      * it observes an event from the data source and assign it as the event
      * timestamp.
      * <p>
+     * With this mode the event time advances even in the absence of events. If
+     * the timestamp is extracted from the event and there are no more events,
```

Comment:
"If the timestamp is extracted from the event and there are no more event" sounds a bit confusing since it sounds like it's referring to this method. I think it would be better to link to the other methods, i.e. if nativeTimestamps or withTimestamps method is used..